tests/Tests/ORM/Functional/GetReferenceOnRelationAsIdTest.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\ORM\EntityNotFoundException;
+use Doctrine\Tests\Models\RelationAsId\Group;
+use Doctrine\Tests\Models\RelationAsId\Membership;
+use Doctrine\Tests\Models\RelationAsId\Profile;
+use Doctrine\Tests\Models\RelationAsId\User;
+use Doctrine\Tests\OrmFunctionalTestCase;
+use Doctrine\Tests\Proxies\__CG__\Doctrine\Tests\Models\RelationAsId\Membership as MembershipProxy;
+use Doctrine\Tests\Proxies\__CG__\Doctrine\Tests\Models\RelationAsId\User as UserProxy;
+
+class GetReferenceOnRelationAsIdTest extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        $this->useModelSet('relation_as_id');
+
+        parent::setUp();
+
+        $u       = new User();
+        $u->id   = 1;
+        $u->name = 'Athos';
+
+        $p       = new Profile();
+        $p->user = $u;
+        $p->url  = 'https://example.com';
+
+        $g       = new Group();
+        $g->id   = 11;
+        $g->name = 'Mousquetaires';
+
+        $m        = new Membership();
+        $m->user  = $u;
+        $m->group = $g;
+        $m->role  = 'cadet';
+
+        $u2       = new User();
+        $u2->id   = 2;
+        $u2->name = 'Portos';
+
+        $this->_em->persist($u);
+        $this->_em->persist($p);
+        $this->_em->persist($g);
+        $this->_em->persist($m);
+        $this->_em->persist($u2);
+        $this->_em->flush();
+        $this->_em->clear();
+    }
+
+    public function testCanGetByValue(): void
+    {
+        $profile = $this->_em->getReference(Profile::class, 1);
+
+        if ($this->_em->getConfiguration()->isLazyGhostObjectEnabled()) {
+            self::assertInstanceOf(UserProxy::class, $profile->user);
+            self::assertEquals('Athos', $profile->user->name);
+        } else {
+            if (PHP_VERSION_ID >= 80100) {
+                $this->markTestIncomplete();
+            }
+            self::assertEquals(1, $profile->user);
+        }
```

Comment:
I figured out that Proxies behave slightly differently [when "Lazy Ghost Objects" are not enabled](https://github.com/doctrine/orm/blob/2.20.x/src/Proxy/ProxyFactory.php#L206). Through experimentation I reached the conclusion that only PHP < 8.1 appears to be displaying the legacy behaviour. [This deprecation notice](https://github.com/doctrine/orm/blob/2.20.x/src/Proxy/ProxyFactory.php#L164) seems to confirm my hypothesis.